esmvalcore/preprocessor/_area.py

```diff
@@ -221,14 +233,19 @@ def zonal_statistics(
             "Zonal statistics on irregular grids not yet implemented"
         )
     (agg, agg_kwargs) = get_iris_aggregator(operator, **operator_kwargs)
-    cube = cube.collapsed('longitude', agg, **agg_kwargs)
-    cube.data = cube.core_data().astype(np.float32, casting='same_kind')
-    return cube
+    stat_cube = cube.collapsed('longitude', agg, **agg_kwargs)
+    if normalize_with_stats is not None:
+        result = get_normalized_cube(cube, stat_cube, normalize_with_stats)
+    else:
+        result = stat_cube
```

Comment:
A bit of a nitpick, but I would find the following easier to read
```suggestion
    result = cube.collapsed('longitude', agg, **agg_kwargs)
    if normalize_with_stats is not None:
        result = get_normalized_cube(cube, result, normalize_with_stats)
```